app/src/main/java/org/mozilla/focus/session/SessionNotificationService.kt

```diff
@@ -150,17 +151,26 @@ class SessionNotificationService : Service() {
             val intent = Intent(context, SessionNotificationService::class.java)
             intent.action = ACTION_START
 
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-                context.startForegroundService(intent)
-            } else {
-                context.startService(intent)
-            }
+            // For #2901: The application is crashing due to the service not calling `startForeground`
+            // so this is a speculative fix to decrease the time between starting the service and it
+            // calling `startForeground`
+            ThreadUtils.postToMainThread(Runnable {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    context.startForegroundService(intent)
+                } else {
+                    context.startService(intent)
+                }
+            })
         }
 
         internal fun stop(context: Context) {
             val intent = Intent(context, SessionNotificationService::class.java)
 
-            context.stopService(intent)
+            // We want to make sure we always call `stopService` after `stopService. So we're giving
```

Comment:
super nit: this is acceptable as is but I'd be even more explicit, "So we're putting these actions on the same sequential run queue."

Without being explicit (to be fair, there is some redundancy in doing so), it's hard to know what the "same treatment" was intended to be without looking at both code paths, especially if this code changes much.